pkg/spdx/implementation.go

```diff
@@ -395,84 +331,56 @@ func (di *spdxDefaultImplementation) PullImagesToArchive(
 		}
 	}
 
+	// Download 4 arch at once
+	t := throttler.New(4, len(references))
+
 	for _, refData := range references {
-		if refData.Tag != "" {
-			tagRef, err := name.ParseReference(refData.Tag)
+		go func(r ImageReferenceInfo) {
+			ref, err := name.ParseReference(r.Digest)
 			if err != nil {
-				return nil, fmt.Errorf("parsing reference %s: %w", referenceString, err)
+				t.Done(fmt.Errorf("parsing reference %s: %w", r.Digest, err))
+				return
 			}
 
-			logrus.Infof("Checking the local image cache for %s", refData.Tag)
-
-			img, err := daemon.Image(tagRef)
-			if err != nil {
-				return nil, fmt.Errorf("getting image %s: %w", referenceString, err)
+			d, ok := ref.(name.Digest)
+			if !ok {
+				t.Done(fmt.Errorf("reference is not a tag or digest"))
+				return
 			}
 
-			if size, err := img.Size(); err == nil && size > 0 {
-				logrus.Infof("%s was found in the local image cache", refData.Tag)
-				// This function is not for digests
-				d, ok := tagRef.(name.Tag)
-				if !ok {
-					return nil, fmt.Errorf("reference is not a tag or digest")
-				}
-				var p string
-				ri := strings.Split(d.RepositoryStr(), "/")
-				if len(ri) > 0 {
-					p = fmt.Sprintf("%s_%s_%s", ri[0], ri[1], d.TagStr())
-				} else {
-					p = fmt.Sprintf("%s_%s", ri[0], d.TagStr())
-				}
+			p := strings.Split(d.DigestStr(), ":")
+			tarPath := filepath.Join(path, p[1]+".tar")
```

Comment:
Can we add a check to avoid causing a panic `if len(p) < 2`?

Reply:
Nice catch, thanks @saschagrunert !